### DISCUSSION TITLE
Alarm Base Station Power Status

### DISCUSSION THREAD

**Discussion by patchrick84** [tsightler/ring-mqtt#952] [Feature Requests]
Is it possible to get the power status (AC or battery backup) from the API for use with Home Assistant?  I have an idea for a specific automation in mind that would require a battery powered device, that isn't a UPS, to let HA know that the power has gone out and I thought my trusty Ring base station would be a good candidate.

**Comment by tsightler** [comment]
The info sensor includes all major attributes of the device, including the AC power status, so you can build an automation however you like.  A user on the HA forums contributed this template sensor which creates a binary sensor that toggles on/off based on presence of mains power.

```
# Template binary sensors
- binary_sensor:
    # Electricity Supply binary sensor
    - name: Electricity Supply
      device_class: problem
      icon: >-
        {%- if is_state('binary_sensor.electricity_supply', 'on') -%}
          mdi:transmission-tower-off
        {%- else -%}
          mdi:transmission-tower
        {%- endif -%}
      state: >-
        {%- if (not is_state('sensor.ring_alarm_base_station_info', 'unavailable'))
        and (not is_state('sensor.ring_alarm_base_station_info', 'unknown'))
        and (not is_state_attr('sensor.ring_alarm_base_station_info', 'acStatus', 'ok')) -%}
          on
        {%- else -%}
          off
        {%- endif -%}  
```

**Reply by patchrick84** [reply]
Awesome.  Thanks so much for the quick reply, @tsightler!
